### PR TITLE
[v1.1] Model adaptor refactor and unit test

### PR DESCRIPTION
**Pull request by zhouyu5**
- update dockerfile pytorch-112
- update unet code patch, add some click-to-run scripts, move unet code from model adaptor directory to modelzoo directory
- refactor domain adapter of model adaptor, add corresponding unit test
- fix some bugs

### CODE REVIEW

File: Dockerfile-ubuntu18.04/DockerfilePytorch112 [hunk 1]
```diff
@@ -4,10 +4,10 @@ FROM intel/oneapi-aikit:2022.3.0-devel-ubuntu18.04
 SHELL ["/bin/bash", "-c"]
 WORKDIR /root/
 ENV PATH /opt/intel/oneapi/intelpython/latest/condabin:$PATH
-RUN apt-get update -y && apt-get install -y openssh-server openjdk-8-jdk sshpass numactl vim
+RUN apt-get update -y && apt-get install -y openssh-server openjdk-8-jdk sshpass numactl vim time
 
 SHELL ["conda", "run", "-n", "pytorch-1.12.0", "/bin/bash", "-c"]
-RUN python -m pip install torchsummary easydict sentencepiece opencv-python scikit-image tensorboard timm boto3 ptflops thop sigopt optuna transformers nnunet
+RUN python -m pip install torchsummary easydict sentencepiece opencv-python scikit-image tensorboard timm boto3 ptflops thop sigopt optuna transformers nnunet wandb pytest
```
Comment (XinyaoWa) [inline Dockerfile-ubuntu18.04/DockerfilePytorch112:10]: Not sure whether need to install "pytest"? @zigzagcai

File: e2eAIOK/ModelAdapter/test/transferrable_model_test.py [hunk 2]
```diff
@@ -465,7 +501,157 @@ def test_init_weight(self):
...
+
+        :return:
+        '''
+        kwargs = self._create_kwargs()
+        model = _make_transferrable(**kwargs)
+        model.train()
+
+        source_data, target_data = torch.randn_like(self.input), torch.randn_like(self.input)
+        source_label, target_label = [torch.zeros(self.input_size)], [torch.zeros(self.input_size)]
+        input_sample = (source_data, target_data)
```
Comment (XinyaoWa) [inline e2eAIOK/ModelAdapter/test/transferrable_model_test.py:585]: Can move all adapter specific code to a separate file.

**Review by XinyaoWa — APPROVED** [review]

File: e2eAIOK/ModelAdapter/src/engine_core/adapter/adversarial/DA_Loss.py [hunk 3]
```diff
@@ -120,7 +119,7 @@ def forward(self, f_s: torch.Tensor, f_t: torch.Tensor,
 class CACDomainAdversarialLoss(nn.Module):
     def __init__(self, **kwargs):
         super().__init__()
-        self.loss_weights = kwargs.pop("loss_weights")
+        self.loss_weights = kwargs.pop("loss_weight")
```
Comment (xuechendi) [inline e2eAIOK/ModelAdapter/src/engine_core/adapter/adversarial/DA_Loss.py:122]: I saw in input args, it is still loss_weights? check "run_dist.sh":line 40, is changing to 'loss_weight' a correct fix?

**Comment by zhouyu5** [comment]
@xuechendi Add some description on this PR. Please have a review on this PR, thank you~

**Comment by xuechendi** [comment]
> @xuechendi Add some description on this PR. Please have a review on this PR, thank you~

I added one comment, pls reply on that, remaining parts of the codes looks good to me

File: e2eAIOK/ModelAdapter/src/engine_core/adapter/adversarial/DA_Loss.py (hunk 3 above)
Comment (zhouyu5) [reply e2eAIOK/ModelAdapter/src/engine_core/adapter/adversarial/DA_Loss.py:122]: Yes, it is right. There are just happened to be the same name, but turned out different things, inside the python code, it is called `lose weight`.

**Comment by zhouyu5** [comment]
> > @xuechendi Add some description on this PR. Please have a review on this PR, thank you~
> 
> I added one comment, pls reply on that, remaining parts of the codes looks good to me

I resolve it, as it is just different variable with similar name in shell and python scripts.

File: e2eAIOK/ModelAdapter/src/engine_core/adapter/adversarial/DA_Loss.py (hunk 3 above)
Comment (xuechendi) [reply e2eAIOK/ModelAdapter/src/engine_core/adapter/adversarial/DA_Loss.py:122]: Thanks

**Review by xuechendi — APPROVED** [review]
LGTM

**Pull request merged**